### PR TITLE
feat(appmesh): rename the class HttpRouteMatchMethod to HttpRouteMethod

### PR DESCRIPTION
**Pull request by Seiya6329**
Renaming `HttpRouteMatchMethod` to `HttpRouteMethod`.
Also move it into its own file since this class is going to be used by multiple classes.

BREAKING CHANGE: the class `HttpRouteMatchMethod` has been renamed to `HttpRouteMethod`

----

*By submitting this pull request, I confirm that my contribution is made under the terms of the Apache-2.0 license*

### CODE REVIEW

**Review by skinny85 — CHANGES_REQUESTED** [review]
Code looks good, but I believe you missed updating the ReadMe 🙂.

File: packages/@aws-cdk/aws-appmesh/test/integ.mesh.ts [hunk 1]
```diff
@@ -2,6 +2,7 @@ import * as ec2 from '@aws-cdk/aws-ec2';
 import * as cloudmap from '@aws-cdk/aws-servicediscovery';
 import * as cdk from '@aws-cdk/core';
 import * as appmesh from '../lib/';
+import { HttpRouteMethod } from '../lib/http-route-method';
```
Comment (skinny85) [inline packages/@aws-cdk/aws-appmesh/test/integ.mesh.ts:5]: This actually should not be imported - you should use the qualified import.

File: packages/@aws-cdk/aws-appmesh/test/test.route.ts [hunk 2]
```diff
@@ -3,6 +3,7 @@ import * as cdk from '@aws-cdk/core';
 import { Test } from 'nodeunit';
 
 import * as appmesh from '../lib';
+import { HttpRouteMethod } from '../lib/http-route-method';
```
Comment (skinny85) [inline packages/@aws-cdk/aws-appmesh/test/test.route.ts:6]: This actually should not be imported - you should use the qualified import.

**Review by skinny85 — CHANGES_REQUESTED** [review]

**Comment by Seiya6329** [comment]
> Code looks good, but I believe you missed updating the ReadMe 🙂.

I was thinking this could be my first ever perfect PR :(

**Review by skinny85 — APPROVED** [review]

**Pull request merged**